forest/src/cli/config.rs

```diff
@@ -2,60 +2,19 @@
 // SPDX-License-Identifier: Apache-2.0, MIT
 
 use chain_sync::SyncConfig;
-use directories::ProjectDirs;
 use forest_libp2p::Libp2pConfig;
 use networks::ChainConfig;
-use rpc_client::DEFAULT_PORT;
 use serde::{Deserialize, Serialize};
-use std::net::SocketAddr;
-use std::path::PathBuf;
-use std::str::FromStr;
 use std::sync::Arc;
 
-#[derive(Serialize, Deserialize, PartialEq)]
+use super::miscellaneous::Miscellaneous;
+
+#[derive(Serialize, Deserialize, PartialEq, Default)]
 #[serde(default)]
 pub struct Config {
-    pub data_dir: PathBuf,
-    pub genesis_file: Option<String>,
-    pub enable_rpc: bool,
-    pub rpc_port: u16,
-    pub rpc_token: Option<String>,
-    /// If this is true, then we do not validate the imported snapshot.
-    /// Otherwise, we validate and compute the states.
-    pub snapshot: bool,
-    pub snapshot_height: Option<i64>,
-    pub snapshot_path: Option<String>,
-    /// Skips loading import CAR file and assumes it's already been loaded.
-    /// Will use the cids in the header of the file to index the chain.
-    pub skip_load: bool,
-    pub encrypt_keystore: bool,
-    /// Metrics bind, e.g. 127.0.0.1:6116
-    pub metrics_address: SocketAddr,
+    pub miscellaneous: Miscellaneous,
```

Comment:
As per @elmattic suggestion:

```suggestion
    pub client: Client,
```